src/NewsService.Business/FindNewsCommand.cs

```diff
@@ -16,38 +34,167 @@ public class FindNewsCommand : IFindNewsCommand
+
+        if (response.Message.IsSuccess)
+        {
+          return response.Message.Body.Departments;
+        }
+        else
+        {
+          _logger.LogWarning("Errors while getting departments. Reason: {Errors}",
+            string.Join('\n', response.Message.Errors));
+        }
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, errorMessage);
+      }
+
+      errors.Add(errorMessage);
+
+      return null;
+    }
 
     public FindNewsCommand(
       INewsRepository repository,
       INewsInfoMapper mapper,
-      IHttpContextAccessor httpContextAccessor)
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<IGetDepartmentsRequest> rcGetDepartments,
+      IRequestClient<IGetUsersDataRequest> rcGetUsers,
+      IRequestClient<IGetImagesRequest> rcGetImages,
+      ILogger<GetNewsCommand> logger,
+      IBaseFindRequestValidator baseFindValidator)
     {
       _repository = repository;
       _mapper = mapper;
       _httpContextAccessor = httpContextAccessor;
+      _rcGetDepartments = rcGetDepartments;
+      _rcGetUsers = rcGetUsers;
+      _rcGetImages = rcGetImages;
+      _logger = logger;
+      _baseFindValidator = baseFindValidator;
     }
 
-    public FindResultResponse<NewsInfo> Execute(FindNewsFilter findNewsFilter)
+    public async Task<FindResultResponse<NewsInfo>> Execute(FindNewsFilter findNewsFilter)
     {
       FindResultResponse<NewsInfo> response = new();
 
-      response.Body = _repository
-        .Find(findNewsFilter, out int totalCount)
-        .Select(_mapper.Map)
-        .ToList();
-
-      response.TotalCount = totalCount;
-
-      if (response.Errors.Any())
+      if (!_baseFindValidator.ValidateCustom(findNewsFilter, out List<string> errors))
       {
         _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
 
         response.Status = OperationResultStatusType.Failed;
+        response.Errors = errors;
         return response;
       }
 
+      List<DbNews> dbNewsList = _repository.Find(findNewsFilter, out int totalCount);
+
+      List<Guid> departmentsIds = dbNewsList.Where(d => d.DepartmentId.HasValue).Select(d => d.DepartmentId.Value).Distinct().ToList();
+      List<DepartmentData> departments = await GetDepartments(departmentsIds, response.Errors);
+
+      List<Guid> authorsIds = dbNewsList.Select(a => a.AuthorId).Distinct().ToList();
+      List<UserData> authors = await GetAuthors(authorsIds, response.Errors);
+
+      List<Guid> imagesIds = new();
+      imagesIds.AddRange(authors?.Where(u => u.ImageId.HasValue).Select(u => u.ImageId.Value).ToList());
+      List<ImageData> avatarImages = await GetImages(imagesIds, response.Errors);
+
+      response.Body = dbNewsList.Select(dbNews => _mapper.Map(dbNews, departments, authors, avatarImages)).ToList();
+      response.TotalCount = totalCount;
       response.Status = OperationResultStatusType.FullSuccess;
 
+      if (response.Body == null)
```

Comment:
you have to return not found in line 175 before sending requests to other services

Reply:
done